eras/shelley/impl/src/Cardano/Ledger/Shelley/API/Validation.hs

```diff
@@ -64,7 +84,8 @@ class
     Environment (Core.EraRule "BBODY" era) ~ STS.BbodyEnv era,
     State (Core.EraRule "BBODY" era) ~ STS.BbodyState era,
     Signal (Core.EraRule "BBODY" era) ~ (Block BHeaderView era),
-    ToCBORGroup (TxSeq era)
+    ToCBORGroup (TxSeq era),
+    Coercible (Event (Core.EraRule "BBODY" era)) (STS.BbodyEvent era)
```

Comment:
some instances don't compile when this is an equality constraint, it's necessary

Reply:
Some instances won't compile if this is in equality constraint. This constraint will get floated outside of typeclass. Maybe it could be an `Embed` constraint instead, but the `Coercible` constraint is not superficially more general